include/externals/trove/aos.h

```diff
@@ -67,7 +67,7 @@ struct use_direct {
 template<typename T>
 __device__ typename enable_if<detail::use_shfl<T>::value, T>::type
 load_warp_contiguous(const T* src) {
-    int warp_id = threadIdx.x & WARP_MASK;
+    int warp_id = ((threadIdx.z * blockDim.y + threadIdx.y) * blockDim.x + threadIdx.x) & WARP_MASK;
```

Comment:
Can you create an issue reminding us that we need to re-implement this every time we update our local version of trove? I know that's rare, but may as well make a note of it.

Reply:
Will do. I think @maddyscientist is working on actually getting these changes into trove.